app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
All the websites have a trailing dot. The process of normalisation adds it there.

 

As mentioned earlier, I think there’s some confusion between validation and normalisation. The FQDN has a trailing dot. The domain names we use do not because it is normalised to one with a trailing dot.